File: Day315/shoretstPathFrom1ton.go
package main

import (
	"fmt"
)

func minStep(n int) int {
	if n == 1 {
		return 1
	}

	answer := 0
	for n >= 1 {
		answer += (n%3 + 1)
		n /= 3
	}
	return answer - 2
}

func main() {
	var t, n int
	fmt.Scan(&t)

	for tc := 0; tc < t; tc++ {
		fmt.Scan(&n)
		result := minStep(n)
		fmt.Println(result)
	}
}

File: Day315/shoretstPathFrom1ton.java
package Day315;
//{ Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG
{
    public static void main(String args[])throws IOException
    {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while(t-- > 0)
        {
            int n = Integer.parseInt(read.readLine());

            Solution ob = new Solution();
            System.out.println(ob.minStep(n));
        }
    }
}
// } Driver Code Ends


class Solution{
    public int minStep(int n)
    {
        // code here
        if (n == 1) {
            return 1;
        }
        int answer = 0;
        while (n >= 1) {
            answer += (n%3+1);
            n/=3;
        }
        return answer - 2;
    }
}
